Sources/SwiftLexicalLookup/IdentifiableSyntax.swift

```diff
@@ -0,0 +1,38 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+/// Syntax node that can be refered to with an identifier.
+public protocol IdentifiableSyntax: SyntaxProtocol {
+  var identifier: TokenSyntax { get }
+}
+
+extension IdentifierPatternSyntax: IdentifiableSyntax {}
+
+extension ClosureParameterSyntax: IdentifiableSyntax {
+  @_spi(Experimental) public var identifier: TokenSyntax {
+    secondName ?? firstName
+  }
+}
+
+extension ClosureShorthandParameterSyntax: IdentifiableSyntax {
+  @_spi(Experimental) public var identifier: TokenSyntax {
+    name
+  }
+}
+
+extension ClosureCaptureSyntax: IdentifiableSyntax {
+  @_spi(Experimental) public var identifier: TokenSyntax {
+    expression.as(DeclReferenceExprSyntax.self)!.baseName
```

Comment:
This won't work for all closure captures. For example:

```swift
{ [x=y+1, weak self] in ... }
```

I suspect the right answer is to rework the syntax tree so that the syntax structure always provides something that we can make an IdentifiableSyntax without the `!`.

Reply:
(this doesn't have to happen now)

Reply:
As for now, I left a comment next to the force unwrap so we don't forget about it. Do you have already an idea how could we alter the syntax tree?

Reply:
Yes, I think we want different syntactic forms for captures like `x`, `x=<expr>`, and `weak x`, where the `x` in every case is just the syntax for the capture name. It will make it a lot easier to identify the names introduced by captures.